eclass/kernel-build.eclass

```diff
@@ -369,6 +388,94 @@ kernel-build_src_install() {
 		secureboot_sign_efi_file "${ED}${kernel_dir}/${image_path}"
 	fi
 
+	if [[ ${KERNEL_IUSE_GENERIC_UKI} ]] && use generic-uki; then
+		local image="${ED}${kernel_dir}/${image_path}"
+
+		# NB: if you pass a path that does not exist or is not a regular
+		# file/directory, dracut will silently ignore it and use the default
+		# https://github.com/dracutdevs/dracut/issues/1136
+		> "${T}"/empty-file || die
+		mkdir -p "${T}"/empty-directory || die
+
+		local dracut_modules=(
+			base bash btrfs cifs crypt crypt-gpg crypt-loop dbus dbus-daemon dm
+			dmraid drm dracut-systemd fido2 i18n iscsi fs-lib kernel-modules
+			kernel-network-modules kernel-modules-extra lunmask lvm nbd mdraid
+			modsign network network-manager nfs nvdimm nvmf pcsc pkcs11 plymouth
+			qemu qemu-net resume rngd rootfs-block shutdown systemd
+			systemd-ac-power systemd-ask-password systemd-initrd
+			systemd-integritysetup systemd-pcrphase systemd-sysusers
+			systemd-udevd systemd-veritysetup terminfo tpm2-tss udev-rules
+			uefi-lib usrmount virtiofs
+		)
+
+		local dracut_args=(
+			--conf "${T}/empty-file"
+			--confdir "${T}/empty-directory"
+			--kernel-image "${image}"
+			--kmoddir "${ED}/lib/modules/${dir_ver}"
+			--kver "${dir_ver}"
+			--verbose
+			--no-hostonly
+			--no-hostonly-cmdline
+			--no-hostonly-i18n
+			--no-machineid
+			--nostrip
+			--no-uefi
+			--early-microcode
+			--reproducible
+			--ro-mnt
+			--modules "${dracut_modules[*]}"
+		)
+
+		# Tries to update ld cache
+		addpredict /etc/ld.so.cache~
+		dracut "${dracut_args[@]}" "${image%/*}/initrd" \
+			|| die "Failed to generate initramfs"
+
+		local ukify_args=(
+			--linux="${image}"
+			--initrd="${image%/*}/initrd"
+			--cmdline="root=/dev/gpt-auto-root ro quiet splash"
+			--uname="${dir_ver}"
+			--output="${image%/*}/uki.efi"
+		)
+
+		if [[ ${KERNEL_IUSE_SECUREBOOT} ]] && use secureboot; then
+			ukify_args+=(
+				--signtool=sbsign
+				--secureboot-private-key="${SECUREBOOT_SIGN_KEY}"
+				--secureboot-certificate="${SECUREBOOT_SIGN_CERT}"
+			)
+			if [[ ${SECUREBOOT_SIGN_KEY} == pkcs11:* ]]; then
+				ukify_args+=(
+					--signing-engine="pkcs11"
+				)
+			else
+				# Sytemd-measure does not currently support pkcs11
+				ukify_args+=(
+					--measure
+					--pcrpkey="${ED}${kernel_dir}/certs/signing_key.x509"
+					--pcr-private-key="${SECUREBOOT_SIGN_KEY}"
+					--phases="enter-initrd"
+					--pcr-private-key="${SECUREBOOT_SIGN_KEY}"
+					--phases="enter-initrd:leave-initrd
+							enter-initrd:leave-initrd:sysinit
+							enter-initrd:leave-initrd:sysinit:ready"
+				)
+			fi
+		fi
+
+		# Systemd<255 does not install ukify in /usr/bin
+		PATH="${PATH}:${BROOT}/usr/lib/systemd:${BROOT}/lib/systemd" \
+			ukify build "${ukify_args[@]}" || die "Failed to generate UKI"
+
+		# Remove kernel image and initrd to save space in the gpkg
+		# We can extract these again later
+		rm "${image}" "${image%/*}/initrd" || \
+			die "Failed to remove kernel image or initramfs"
```

Comment:
```suggestion
		rm "${image}" "${image%/*}/initrd" || die
```

I don't think there's really a point in adding an explicit message when the "die line" clearly indicates what failed.